docs/releases/3.0.txt

```diff
@@ -525,8 +525,8 @@ Miscellaneous
 * Support for ``pywatchman`` < 1.2.0 is removed.
 
 * HTML rendered by :class:`~django.forms.SelectDateWidget` for required fields
-  now have the ``placeholder`` attribute, which, mainly, may require some
-  adjustments in tests that compare HTML.
+  now includes the ``required`` attribute when an ``empty_label`` is specified,
```

Comment:
This looks backwards to me. Just because I specified `empty_label` doesn't mean I want the `select` to be `required`. (Indeed how else do I have a non-required select except to add an `empty_label`?)

Should the logic not be:

> ... now includes an ``empty_label`` when the ``required`` attribute is specified.

?

Reply:
The intention was to have the logic be "if empty_label is specified and is_required is set" but then I realized we don't inject the empty option if the field is marked as required. That behaviour is different from the `ModelChoice` fields where if you specify `empty_label`, the empty option is included by default.